glean-core/src/upload/directory.rs

```diff
@@ -19,6 +19,23 @@ use crate::{DELETION_REQUEST_PINGS_DIRECTORY, PENDING_PINGS_DIRECTORY};
 /// this will contain the document_id, path, JSON encoded body of a ping and the persisted headers.
 pub type PingPayload = (String, String, String, Option<HeaderMap>);
 
+/// A struct to hold the result of scanning all pings directories.
+#[derive(Clone, Debug, Default)]
+pub struct PingPayloadsByDirectory {
+    pub pending_pings: Vec<(fs::Metadata, PingPayload)>,
```

Comment:
Do we actually need the full metadata? I think we only use the size.
`fs::Metadata` [is a wooping 176 bytes](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6e574ebe39ec305cd963d67808807b8b) (on Linux at least).

Reply:
Not really, we only use the size. I can change that.